src/main/java/org/opensrp/repository/postgres/LocationTagRepositoryImpl.java

```diff
@@ -225,4 +233,96 @@ private LocationTag convert(org.opensrp.domain.postgres.LocationTag pgLocationTa
 		return convert(pgLocationTagList);
 	}
 	
+	private LocationTagMap convertLocationTagMap(org.opensrp.domain.postgres.LocationTagMap pgLocationTagMap) {
+		if (pgLocationTagMap == null) {
+			return null;
+		}
+		LocationTagMap locationTagmap = new LocationTagMap();
+		
+		locationTagmap.setLocationId(pgLocationTagMap.getLocationId());
+		locationTagmap.setLocationTagId(pgLocationTagMap.getLocationTagId());
+		
+		return locationTagmap;
+	}
+	
+	private org.opensrp.domain.postgres.LocationTagMap convertLocationTagMap(LocationTagMap locationTagMap) {
+		if (locationTagMap == null) {
+			return null;
+		}
+		org.opensrp.domain.postgres.LocationTagMap pgLocationTagMap = new org.opensrp.domain.postgres.LocationTagMap();
+		pgLocationTagMap.setLocationId(locationTagMap.getLocationId());
+		pgLocationTagMap.setLocationTagId(locationTagMap.getLocationTagId());
+		
+		return pgLocationTagMap;
+	}
+	
+	private List<LocationTagMap> convertLocationTagMap(List<org.opensrp.domain.postgres.LocationTagMap> pgLocationTagMaps) {
+		List<LocationTagMap> locationTagMaps = new ArrayList<>();
+		if (isEmptyList(pgLocationTagMaps)) {
+			return locationTagMaps;
+		}
+		for (org.opensrp.domain.postgres.LocationTagMap pgLocationTagMap : pgLocationTagMaps) {
+			locationTagMaps.add(convertLocationTagMap(pgLocationTagMap));
+		}
+		return locationTagMaps;
+	}
+	
+	@Override
+	public int addLocationTagMap(LocationTagMap locationTagMap) {
+		org.opensrp.domain.postgres.LocationTagMap pgLocationTagMap = convertLocationTagMap(locationTagMap);
+		if (locationTagMap == null) {
+			throw new IllegalArgumentException("location tag map not specified");
+		}
+		if (locationTagMap.getLocationId() == null) {
+			throw new IllegalArgumentException("location id not specified");
+		}
+		if (locationTagMap.getLocationTagId() == null) {
+			throw new IllegalArgumentException("location tag id not specified");
+		}
```

Comment:
This is usually handled by the examples on the generated code if you specify that these are not null on the schema definition